pkg/service/pgsvc/contract.go

```diff
@@ -93,238 +178,258 @@ func (s *ContractSvc) Add(ctx context.Context, contract *model.Contract) (*model
 		}
+	}
 
-		if e := validator(c); e != nil {
-			return e
+	if !common.IsHexAddress(contractAddress) {
+		return nil, &model.BackendError{
+			Cause:   model.ErrValidationFailed,
+			Message: model.ValidationErrorInvalidFormat("contract_address"),
 		}
+	}
 
-		o, err := queries.ContractPatch(ctx, *patchParams)
-
-		if errors.Is(err, sql.ErrNoRows) {
-			return model.ErrEntityNotFound
+	return s.toStatus(ctx, actorID, &pgdao.ContractPatchParams{
+		StatusChange:          true,
+		Status:                targetStatus,
+		ContractAddressChange: true,
+		ContractAddress:       contractAddress,
+		ID:                    id,
+	}, func(c *model.ContractDTO) error {
+		if c.CustomerID != actorID {
+			return model.ErrInsufficientRights
 		}
 
-		result = &model.Contract{ //nolint: dupl
-			ID:               o.ID,
-			Title:            o.Title,
-			Description:      o.Description,
-			Price:            decimal.RequireFromString(o.Price),
-			Duration:         o.Duration.Int32,
-			Status:           o.Status,
-			CreatedAt:        o.CreatedAt,
-			UpdatedAt:        o.UpdatedAt,
-			CreatedBy:        o.CreatedBy,
-			ContractAddress:  o.ContractAddress,
-			CustomerAddress:  o.CustomerAddress,
-			PerformerAddress: o.PerformerAddress,
+		if c.Status != allowedSourceStatus {
+			return fmt.Errorf("%w: unable to move from %s to %s", model.ErrInappropriateAction, c.Status, targetStatus)
 		}
 
-		return err
+		return nil
 	})
 }
 
-// Accept makes contract accepted if any
-func (s *ContractSvc) Accept(ctx context.Context, id, actorID, performerAddress string) (*model.Contract, error) {
-	allowedSourceStatus := model.ContractCreated
-	targetStatus := model.ContractAccepted
+// Sign makes contract signed
+func (s *ContractSvc) Sign(ctx context.Context, id, actorID string) (*model.ContractDTO, error) {
+	allowedSourceStatus := model.ContractDeployed
+	targetStatus := model.ContractSigned
+
 	return s.toStatus(ctx, actorID, &pgdao.ContractPatchParams{
-		StatusChange:           true,
-		Status:                 targetStatus,
-		PerformerAddressChange: true,
-		PerformerAddress:       performerAddress,
-		ID:                     id,
-	}, func(c *model.Contract) error {
-		if c.Performer.ID != actorID {
+		StatusChange: true,
+		Status:       targetStatus,
+		ID:           id,
+	}, func(c *model.ContractDTO) error {
+		if c.PerformerID != actorID {
 			return model.ErrInsufficientRights
 		}
-		if c.Status != allowedSourceStatus {
-			return fmt.Errorf("%w: unable to move from %s to %s", model.ErrInappropriateAction, c.Status, targetStatus)
-		}
 
-		if performerAddress == c.CustomerAddress {
+		if !common.IsHexAddress(c.ContractAddress) {
 			return &model.BackendError{
-				Cause:    model.ErrInsufficientFunds,
-				Message:  "customer and performer addresses cannot be the same",
-				TechInfo: "performer_address",
+				Cause:    model.ErrValidationFailed,
+				Message:  model.ValidationErrorInvalidFormat("contract_address"),
+				TechInfo: c.ContractAddress,
```

Comment:
Should be here name of the field, not value?

Reply:
It would be good to talk about it later